src/gululog_w_cur.erl

```diff
@@ -106,6 +108,22 @@ truncate(Dir, Cur, SegId, Position, BackupDir) ->
   TruncateResult = truncate_truncate_do(TruncateFile, Position, BackupDir),
   {open_do(Dir, KeepFiles), DeleteResult ++ TruncateResult}.
 
+%% @doc Delete segment file depends on given segid
+%% return the deleted segid
+%% @end
+-spec delete_seg(dirname(), segid()) -> segid().
+delete_seg(Dir, SegId) ->
+  delete_seg(Dir, SegId, ?undef).
+
+%% @doc Delete segment file depends on given segid
+%% return the deleted segid
+%% backup the segment file when the given argument 'backupdir' is not '?undef'
+%% @end
+-spec delete_seg(dirname(), segid(), ?undef | dirname()) -> segid().
+delete_seg(Dir, SegId, BackupDir) ->
```

Comment:
should accept the #wcur{} structure as the second argument, two reasons:
1. We should assert the the one being deleted is not Cur#wcur.segid
2. To force the API usage for gululog_topic only.